lib/workers/global/config/parse/env.ts

```diff
@@ -118,102 +118,92 @@ function massageConvertedExperimentalVars(
 
 export async function getConfig(
   inputEnv: NodeJS.ProcessEnv,
+  configEnvKey = 'RENOVATE_CONFIG',
 ): Promise<AllConfig> {
-  let env = inputEnv;
-  env = normalizePrefixes(inputEnv, inputEnv.ENV_PREFIX);
+  let env = normalizePrefixes(inputEnv, inputEnv.ENV_PREFIX);
   env = massageConvertedExperimentalVars(env);
   env = renameEnvKeys(env);
   // massage the values of migrated configuration keys
   env = massageEnvKeyValues(env);
 
   const options = getOptions();
 
-  let config: AllConfig = {};
+  const config = await parseAndValidateOrExit(env, configEnvKey);
 
-  if (env.RENOVATE_CONFIG) {
-    try {
-      config = JSON5.parse(env.RENOVATE_CONFIG);
-      logger.debug({ config }, 'Detected config in env RENOVATE_CONFIG');
+  config.hostRules ??= [];
 
-      config = await migrateAndValidateConfig(config, 'RENOVATE_CONFIG');
-    } catch (err) {
-      logger.fatal({ err }, 'Could not parse RENOVATE_CONFIG');
-      process.exit(1);
+  for (const option of options) {
+    if (option.env === false) {
+      continue;
     }
-  }
 
-  config.hostRules ??= [];
+    const envName = getEnvName(option);
+    const envVal = env[envName];
+    if (!envVal) {
+      continue;
```

Comment:
the diff is missleading,
the only thing changed apart of the `parseAndValidateOrExit` function is converting foreach to for ... of.
\+ flip if statements so that we can early exit (continue here), so that we can have less indentation.